### PR TITLE
Fix: convert NaN and NaT to None for test generation

### LINKED ISSUE #2436: [dbt integration][unit test][bug?] `sqlmesh create_test` fails on timestamp column with duckdb
**Context**

- sqlmesh: 0.87.0
- duckdb: 0.9.2
- dbt-core: 1.7.11
- dbt-duckdb: 1.7.3

**What's the issue**

On a dbt project, using dbt integration, running `sqlmesh create_test` throw an error on a timestamp column:
`ruamel.yaml.representer.RepresenterError: cannot represent an object: NaT`

**How to reproduce**

1. Clone https://github.com/bqbooster/jaffle_shop_duckdb
2. `cd base`
3.  `DBT_PROFILES_DIR=. sqlmesh plan` (run the full plan to build the tables)
4. 
```
DBT_PROFILES_DIR=. sqlmesh create_test dbt_cou.customers \
--query dbt_cou.stg_customers "SELECT * FROM dbt_cou.stg_customers LIMIT 3" \
--query dbt_cou.stg_orders "SELECT order_id, customer_id, CURRENT_DATE() as order_date, status FROM dbt_cou.stg_orders LIMIT 3" \
--query dbt_cou.stg_payments "SELECT * FROM dbt_cou.stg_payments LIMIT 3" \
--overwrite
```

=> it fails with following stacktrace:

```
File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/sqlmesh/cli/main.py", line 501, in create_test
    obj.create_test(
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/sqlmesh/core/context.py", line 1332, in create_test
    generate_test(
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/sqlmesh/core/test/definition.py", line 569, in generate_test
    yaml.dump({test_name: test_body}, file)
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/sqlmesh/utils/yaml.py", line 60, in dump
    yaml.YAML().dump(value, stream or result)
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/main.py", line 594, in dump
    return self.dump_all([data], stream, transform=transform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/main.py", line 604, in dump_all
    self._context_manager.dump(data)
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/main.py", line 953, in dump
    self._yaml.representer.represent(data)
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 81, in represent
    node = self.represent_data(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 103, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 316, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 839, in represent_mapping
    node_value = self.represent_data(item_value)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 103, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 316, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 839, in represent_mapping
    node_value = self.represent_data(item_value)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 103, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 316, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 839, in represent_mapping
    node_value = self.represent_data(item_value)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 103, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 307, in represent_list
    return self.represent_sequence('tag:yaml.org,2002:seq', data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 748, in represent_sequence
    node_item = self.represent_data(item)
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 103, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 316, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 839, in represent_mapping
    node_value = self.represent_data(item_value)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 113, in represent_data
    node = self.yaml_representers[None](self, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/anaconda3/envs/dbt-1.7/lib/python3.11/site-packages/ruamel/yaml/representer.py", line 345, in represent_undefined
    raise RepresenterError(f'cannot represent an object: {data!s}')
```

Note: I tried to select `order_date` directly but it's still the same issue.

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #2436

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/test/definition.py [hunk 1]
```diff
@@ -544,10 +545,14 @@ def generate_test(
...
-                ctes[cte.alias] = pandas_timestamp_to_pydatetime(
-                    cte_output.apply(lambda col: col.map(_normalize_dataframe)),
-                    cte_query.named_selects,
-                ).to_dict(orient="records")
+                ctes[cte.alias] = (
+                    pandas_timestamp_to_pydatetime(
+                        cte_output.apply(lambda col: col.map(_normalize_df_value)),
+                        cte_query.named_selects,
+                    )
+                    .replace({np.nan: None})
```
Comment (tobymao) [inline sqlmesh/core/test/definition.py:553]: i think these replaces will trigger a warning, i've seen it in sqlglot

Comment (georgesittas) [reply sqlmesh/core/test/definition.py:553]: I'm not sure if this is the direction that warns, maybe it's just `None -> np.nan`. Looking into it now.

**Comment by github-christophe-oudar** [comment]
I can confirm that the fix works 👍

File: sqlmesh/core/test/definition.py (hunk 1 above)
Comment (georgesittas) [reply sqlmesh/core/test/definition.py:553]: Brought back the `np.nan -> None` mapping, couldn't repro https://github.com/TobikoData/sqlmesh/issues/1799. May leave the PR open for a bit to evaluate this change and check if there are warnings.

**Pull request merged**